packages/core/src/button/Button.tsx

```diff
@@ -26,6 +27,19 @@ export interface ButtonProps extends ComponentPropsWithoutRef<"button"> {
    * 'primary' is the default value.
    */
   variant?: ButtonVariant;
+  /**
+   * To show a loading spinner.
+   */
+  isLoading?: boolean;
+  /**
+   * For the screen reader to announce while button is in loading state
+   * 'Loading' is the default value.
+   */
+  loadingText?: string;
```

Comment:
Do we really need a new prop to deal with announcement? 
cc @joshwooding

Reply:
Without clear accessibility guidance I don't want any of these added apart from the isLoading prop

Reply:
Now we see this in code ...Should this move to it's own component or are we comfortable having one API inside another ? On other component library's they just have one prop which drives the state.